Update docs building and actions

abhro opened the pull request:
- Use newer Documenter
- Use GitHub Actions to deploy docs
- Fix doctests

abhro committed:
Reflow CompatHelper workflow file

abhro committed:
Use julia-actions/cache for action caching

abhro committed:
Update codecov-action version

abhro committed:
Create Documentation.yml

abhro committed:
Add self dep to docs/Project.toml

abhro committed:
Bump Documenter.jl dep version

abhro committed:
Separate out badges

abhro committed:
Update syntax highlight tag

abhro committed:
Add syntax highlight tag

abhro committed:
Make markdown list in docstring

abhro committed:
Add doc page header (and title)

abhro committed:
Add module to autodocs block

abhro committed:
Remove deps option from deploydocs

abhro committed:
Fix doctests

abhro committed:
Share doctest environment

abhro commented:
gentle bump (@timholy?)

oxinabox commented:
I am kinda surprised this package is still working.
Since most newer work is on DimensionalData.jl
or perhaps KeyedArrays.jl or one of the others.
 
 But still this PR looks sound.
 And having the docs work is good.
 
Ping me to merge this if i forget to once CI is passing

abhro commented:
Unit tests fail on newest Julia v1 (1.12.4) :/
The most recent commit on main (https://github.com/JuliaArrays/AxisArrays.jl/commit/91eeb428585894e4e0fc053676093a679d6ab778) also seems to fail on those tests. The logs for that are expired, though, so I can't tell if it's failing in the same manner

abhro committed:
Update CI testing matrix in UnitTest.yml

Put julia-version on top so that it matches the outer matrix, and so the reported job name is in the same format for all of them

Signed-off-by: abhro <5664668+abhro@users.noreply.github.com>

abhro commented:
The test failure is on `combine()`, which this PR doesn't change. Should this be considered good to merge?
@oxinabox
